### ISSUE TITLE
ios_framework_binary: can't use Objective-C definitions from Swift

### ISSUE THREAD

**Issue by Ubehebe** [bazelbuild/bazel#875]
I'm trying to create a framework that contains both Objective-C and Swift sources, where the Swift sources use Objective-C symbols. Apple's directions for doing so are [here](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html) (see the section "Importing Code from Within the Same Framework Target").

As far as I can tell, there is no way to get ios_framework_binary to do this.

Minimal repro case:

BUILD:

``` python
ios_framework_binary(
    name = "Framework",
    hdrs = [
        "Foo.h",
    ],
    srcs = [
        "Foo.m",
        "Bar.swift",
    ],
    enable_modules = 1,
)
```

Foo.h:

``` objc
#import <Foundation/Foundation.h>
@interface Foo: NSObject
@end
```

Foo.m:

``` objc
#import "Foo.h"
@implementation Foo
@end
```

Bar.swift:

``` swift
class Bar {
  func baz(foo: Foo /* defined in Foo.h */) {}
}
```

`bazel build :Framework` gives the following error:

```
Bar.swift:2:17: error: use of undeclared type 'Foo'
```

This makes sense; we haven't done anything to tell the Swift sources about the Objective-C definitions.
Apple's directions for doing this within the same framework target are (see link above):

```
To import a set of Objective-C files in the same framework target as your Swift code, you’ll need to import those files into the Objective-C umbrella header for the framework.
```

[ios_framework_binary](http://bazel.io/docs/be/objective-c.html#ios_framework_binary) doesn't have any options dealing with umbrella headers. (ios_framework does have a `hdrs` option that looks promising, but ios_framework requires an ios_framework_binary to build.)

ios_framework_binary does have a `bridging_header` option: "A header defining the Objective-C interfaces to be exposed in Swift." So let's try that (even though Apple's directions talk about bridging headers only in the context of an app target, not a framework). We add the following to the BUILD file:

``` python
bridging_header = "Framework.h",
```

Framework.h:

``` objc
#import "Foo.h"
```

Now `bazel build :Framework` causes xcrunwrapper to segfault.

So in short, I haven't figured out how to get a mixed Objective-C/Swift ios_framework_binary target to build. Any help is appreciated.

**Comment by dmishe** [comment]
This is definitely a problem on multiple levels and I am afraid I don't have a good fix for this at the moment.

Native swift support is going away in favor of the swift_library rule, and with that you can't have Objective-C and Swift code inside of the same target. Instead you compose binaries out of objc_/swift_ libraries. That works well for ios_application, but may not work for ios_framework. I will need to look into frameworks support with the new rule.

**Comment by dmishe** [comment]
@Ubehebe what is your intended use case here, do you want to publish this framework or simply package it in your app?

**Comment by keith** [comment]
When running bazel from master `.swift` files now seem to be disallowed in `ios_framework_binary`

**Comment by dmishe** [comment]
That's right, and ios_framework is broken anyway. @c-parsons knows more about a proper solution.

**Comment by c-parsons** [comment]
ios_framework{,binary} are deprecated and on the way out -- we're cooking up a skylark solution to framework generation that will replace ios_framework entirely. ETA about one month.

**Comment by afking** [comment]
@c-parsons any updates on the new skylark solution?

**Comment by c-parsons** [comment]
It is done!
https://github.com/bazelbuild/rules_apple
